backend/packages/Upgrade/src/api/services/MetricService.ts

```diff
@@ -62,7 +64,7 @@ export class MetricService {
   }
 
   private async checkMetricsPermission(): Promise<boolean> {
-    const setting = await this.settingService.getClientCheck();
+    const setting = await this.settingService.getClientCheck(new UpgradeLogger());
```

Comment:
Why are you creating new logger in the service?

Reply:
it is required as we have logger instance inside the function